hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1500,6 +1506,22 @@ void QuectelNcpClient::checkRegistrationState() {
 int QuectelNcpClient::interveneRegistration() {
     CHECK_TRUE(connState_ == NcpConnectionState::CONNECTING, SYSTEM_ERROR_NONE);
 
+    // Check and store if we are using a Twilio SIM based on ICCID
+    if (!iccidChecked_) {
```

Comment:
Nitpicking: Should we move this `if` condition to a separate function for reusability? This can be a general utility to set `iccidChecked_` instead of being only in `interveneRegistration()`

Reply:
Resolving this one as it's duplicated :D